src/repr/src/row.rs

```diff
@@ -591,13 +645,59 @@ impl Row {
         Ok(packer.finish())
     }
 
+    /// Allocate an empty `Row` with a pre-allocated capacity.
+    #[inline]
+    pub fn with_capacity(cap: usize) -> Self {
+        Self {
+            data: SmallVec::with_capacity(cap),
+        }
+    }
+
+    /// Extend an existing `Row` with a `Datum`.
+    #[inline]
+    pub fn push<'a, D>(&mut self, datum: D)
```

Comment:
Yeah, fair point. I think it maybe checks out for the moment, in that the uses of `.push_dict` are places where we don't know the final size ahead of time, and we would want to pilot folks to use `RowPacker` probably? There is a .. idk minor risk with these too that if things error we want to avoid leaving a half-formed `Row` that would be unsafe to unpack; that is where the UB lives in this module, I think.

Reply:
Mmh, if only typestates were more accessible; I think this would make a good use case, but that would require a massive re-do of all our row constructions (:

Reply:
I think if `Row` is mutable, then you could use a `Row` as a `RowPacker`. The only Q I don't know the answer to is if `RowPacker` can get in a bad state with erroring packing, in which case it is nice to have it be in a non-`unpack`able state.  But, I'm not up to speed on that at the moment.